week1/homework/src/server.js

```diff
@@ -10,6 +10,32 @@ function createServer(port) {
 
   const server = http.createServer((request, response) => {
     // TODO: Write your homework code here
+
+    switch (request.url) {
+      case '/state':
+        serverResponse();
+        break;
+      case '/add':
+        state += 1;
+        serverResponse();
+        break;
+      case '/subtract':
+        state -= 1;
+        serverResponse();
+        break;
+      case '/reset':
+        state = 10;
+        serverResponse();
+        break;
+      default:
+        response.writeHead(404, { 'Content-Type': 'application/json' });
+        response.end(JSON.stringify({ error: 'Not found' }));
+        break;
+    }
+    function serverResponse() {
```

Comment:
You shouldn't define functions inside of other functions. `serverResponse` here gets defined every time we call `http.createServer`.